crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -180,7 +186,15 @@ impl CheckpointArtifacts {
             })
     }
 
+    pub fn is_empty(&self) -> bool {
+        self.artifacts.iter().all(|artifact| artifact.is_empty())
+    }
+
     pub fn digest(&self) -> SuiResult<CheckpointArtifactsDigest> {
+        if self.is_empty() {
+            return Err(SuiError::from("Cannot compute digest for empty artifacts"));
+        }
+
         // Already sorted by BTreeSet!
         let digests = self
             .artifacts
```

Comment:
you probably also want to skip any artifacts that are empty to prevent them being added to the calculation of the hash

Reply:
hmm, this one i am less sure of since the artifacts digest size doesn't change. do you feel strongly that we shouldn't include empty artifact's? 

the reason i am hesitant is that it complicates the light-client logic, i.e., the logic to prove that an object wasn't modified in a checkpoint. this is true even of the change i already made (but there i suppose the savings in checkpoint summary sizes is important enough).

Reply:
it may not impact the size, but it does impact the calculation of the digest itself